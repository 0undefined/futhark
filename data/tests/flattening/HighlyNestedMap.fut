// --
// input {
//   [ [ [ [1,2,3], [4,5,6] ]
//     , [ [6,7,8], [9,10,11] ]
//     ]
//   , [ [ [3,2,1], [4,5,6] ]
//     , [ [8,7,6], [11,10,9] ]
//     ]
//   ]
//   [ [ [ [4,5,6]  , [1,2,3] ]
//     , [ [9,10,11], [6,7,8] ]
//     ]
//   , [ [ [4,5,6]  , [3,2,1] ]
//     , [ [11,10,9], [8,7,6] ]
//     ]
//   ]
// }
// output {
//   [[[[5, 7, 9],
//      [5, 7, 9]],
//     [[15, 17, 19],
//      [15, 17, 19]]],
//    [[[7, 7, 7],
//      [7, 7, 7]],
//     [[19, 17, 15],
//      [19, 17, 15]]]]
// }
fun [int] add1 ([int] xs, [int] ys) =
  map(+, zip (xs,ys))

fun [[int]] add2 ([[int]] xs, [[int]] ys) =
  map (add1, zip (xs,ys))

fun [[[int]]] add3 ([[[int]]] xs, [[[int]]] ys) =
  map (add2, zip (xs,ys))

fun [[[[int]]]] add4 ([[[[int]]]] xs, [[[[int]]]] ys) =
  map (add3, zip (xs,ys))

fun [[[[int]]]] main([[[[int]]]] A, [[[[int]]]] B) =
  add4(A,B)
