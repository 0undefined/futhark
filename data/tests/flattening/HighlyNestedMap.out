[[[[5, 7, 9],
   [5, 7, 9]],
  [[15, 17, 19],
   [15, 17, 19]]],
 [[[7, 7, 7],
   [7, 7, 7]],
  [[19, 17, 15],
   [19, 17, 15]]]]
